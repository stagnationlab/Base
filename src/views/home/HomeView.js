import classNames from 'classnames';
import React, { Component } from 'react';
import './HomeView.scss';

export default class HomeView extends Component {
	
	static defaultProps = {};

	constructor(props) {
		super(props);
		this.state = {
			isOkToShowThis: true,
		};
		this.minutv = 0;
	}

	render() {
		const {
			someProp,
			anotherThing,
		} = this.props;

		const {
			isOkToShowThis,
		} = this.state;

		const btnClassName = 'btn123';
		const className = classNames(
			'HomeView babelPlugin babelPlugin babelPlugin',
			btnClassName,
			{
				isActive: this.isActive(),
				isVisible: isOkToShowThis,
			}
		);

		return (
			<div className={className} onClick={this.handleClick}>
				tere
			</div>
		);
	}


	handleClick(e) {
		e.preventDefault();
	}

	isActive() {
		return Math.random() > 0.5;
	}
}