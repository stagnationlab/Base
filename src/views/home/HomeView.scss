.HomeView {
	&-post {
		background-color: #ccc;
		overflow: hidden;
		margin-bottom: 20px;

		img {
			max-width: 400px;
			float: left;
			margin: 20px;
		}
	}
}