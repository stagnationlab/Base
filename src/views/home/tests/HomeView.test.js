import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import { HomeView } from '../HomeView';

const props = {
	error: null,
	isLoading: false,
	posts: [
		{
			id: 1,
			title: 'test 1',
			body: 'test content 1',
		},
		{
			id: 2,
			title: 'test 2',
			body: 'test content 2',
		},
	],
	getPostsByUserId: jest.fn(),
};

const wrapper = mount(
	<MemoryRouter>
		<HomeView {...props} />
	</MemoryRouter>,
);

describe('HomeView', () => {
	it('renders correctly', () => {
		expect(toJson(wrapper.find('HomeView'))).toMatchSnapshot();
	});
});