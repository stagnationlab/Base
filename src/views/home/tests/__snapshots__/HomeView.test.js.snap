// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView renders correctly 1`] = `
<HomeView
  error={null}
  getPostsByUserId={[Function]}
  isLoading={false}
  posts={
    Array [
      Object {
        "body": "test content 1",
        "id": 1,
        "title": "test 1",
      },
      Object {
        "body": "test content 2",
        "id": 2,
        "title": "test 2",
      },
    ]
  }
>
  <div
    className="HomeView"
  >
    <h1>
      Posts
    </h1>
    <div
      className="HomeView-post"
    >
      <img
        alt="test 1"
        src="http://lorempixel.com/200/200/animals/1"
      />
      <h2>
        test 1
      </h2>
      <p>
        <Link
          replace={false}
          to="/post/1"
        >
          <a
            href="/post/1"
            onClick={[Function]}
          >
            view more
          </a>
        </Link>
      </p>
    </div>
    <div
      className="HomeView-post"
    >
      <img
        alt="test 2"
        src="http://lorempixel.com/200/200/animals/2"
      />
      <h2>
        test 2
      </h2>
      <p>
        <Link
          replace={false}
          to="/post/2"
        >
          <a
            href="/post/2"
            onClick={[Function]}
          >
            view more
          </a>
        </Link>
      </p>
    </div>
  </div>
</HomeView>
`;
